analyzers/src/SonarAnalyzer.CSharp/Rules/ConditionalStructureSameImplementation.cs

```diff
@@ -78,7 +78,7 @@ protected override void Initialize(SonarAnalysisContext context)
 
                     var precedingSection = switchSection
                         .GetPrecedingSections()
-                        .FirstOrDefault(preceding => CSharpEquivalenceChecker.AreEquivalent(switchSection.Statements, preceding.Statements));
+                        .FirstOrDefault(preceding => CSharpEquivalenceChecker.AreEquivalentWithEqualInvocations(switchSection.Statements, preceding.Statements, c.SemanticModel));
```

Comment:
Any reason to create a separate method for this and not update the existing one?

Reply:
The first reason is that the `(SyntaxList<SyntaxNode> nodeList1, SyntaxList<SyntaxNode> nodeList2)` is actually calling the equivalent common method written for both VB and C# - and I'd need to change the behavior (and also use the semantic model) for other files/rules that I'm not sure if they actually need it at all. 

The second reason is that IMO the method `IsEquivalent` checks if the two groups of statements are equivalent (from a syntax point of view) and not equal - so I'd really change the expected behavior in a weird way if I'd check only for invocations if they are equal.


I'm still not 100% happy with the solution I gave. Any ideas are more than welcome!
Maybe it would be better to move the `AreEquivalentWithEqualInvocations` at least inside the rule file since it's a method written only for this rule.